addon/model.js

```diff
@@ -322,7 +198,7 @@ export default class MegamorphicModel extends EmberObject {
     let oldValue = this._cache[key];
     let newValue = this._internalModel._modelData.getAttr(key);
 
-    let oldIsRecordArray = oldValue && oldValue instanceof M3RecordArray;
+    let oldIsRecordArray = oldValue && oldValue instanceof M3ReferenceArray;
```

Comment:
is there a reason to check against `M3ReferenceArray` instead of `M3RecordArray`?

I think we might want `M3RecordArray` as this will still catch reference arrays, but it will also keep the same functionality for people who `model.set(attr, resultOfQueryURL)`

Reply:
oh ok. I will fix that.